CONTRIBUTING.md

```diff
@@ -39,11 +40,25 @@ stack test
 stack bench
 ```
 
+
 ## Troubleshooting
 
-- If you're using Stack, make sure you have version >= 1.6.1 ([1], [2])
+- If you're using Stack, make sure you have version >= 1.6.1
+  ([stack#3524](https://github.com/commercialhaskell/stack/issues/3624),
+  [stack#3345](https://github.com/commercialhaskell/stack/issues/3345)).
+
+
+## Sending Pull Requests
+
+When you send a pull request, please:
 
+- Link to libraries@haskell.org discussion thread if you are changing the public
+  API.
 
-[1] https://github.com/commercialhaskell/stack/issues/3624
+- Run the benchmarks and include the results in the commit message and Pull
+  Request.
 
-[2] https://github.com/commercialhaskell/stack/issues/3345
+- Add unit tests exercising any changes you make, especially if it is a bug
+  fix.
```

Comment:
QuickCheck properties are more important than unit tests, generally. But if the problem occurs too rarely for QuickCheck to hit it reasonably reliably, a unit test would be a good idea.

Reply:
Done.